rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/SemanticTokenizer.java

```diff
@@ -502,6 +540,22 @@ private static boolean isKeyword(ITree tree) {
 
             return true;
         }
+
+        // Derived from `org.rascalmpl.values.parsetrees.ProductionAdapter.getCategory`
+        private static Mode getModeIfPresent(IConstructor prod, Mode ifAbsent) {
```

Comment:
It might be a good idea to submit a PR for Rascal to refactor `ProductionAdapter.getCategory` and extract `getAttribute(String name)`.

I understand you cannot use that method yet, but after the next release of Rascal you can simplify this code to use the `getAttribute` method.

Reply:
Good suggestion. I ended up not using tags after all, so this code isn't needed anymore, but I filed an [issue](https://github.com/usethesource/rascal/issues/2072) to do this improvement at some point anyway.